ably/ably-chat-js
# react-hooks: Add useMessages hook

### Context

* [CHA-384]
* As part of the react hooks development, a new hook to interact with the `Messages` instance of a room was required.

### Description

* Added `useMessages` hook to provide access to the `Messages` instance. This allows the user to interact with the underlying `Messages` instance, send messages, query historical messages and subscribe to new messages.
  * An optional parameter can be supplied alongside a listener, this will allow messages up-until the point the listener was subscribed.
* Updated the demo app to use the new hook for messages handling and removed the old custom hook.
* Updated the readme to document hook usage.
* Added unit tests for the new hook.
* Added integration tests for the new hook.

### Checklist


### Testing Instructions (Optional)

1. Spin up a local instance of the demo app.
2. Ensure you can send messages.
3. In another window, ensure you receive the messages.
4. Refresh a window and check the historical messages have been loaded.

[CHA-384]: https://ably.atlassian.net/browse/CHA-384?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
